sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,232 @@
+    assert C.masscenter.vel(N).simplify() == (omega * A.z)/sqrt(2)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == \
+        (1 - sqrt(2)/2)*N.x + sqrt(2)*cos(theta)/2*N.y + sqrt(2)*sin(theta)/2*N.z
+    assert C.masscenter.vel(N).express(N).simplify() == - sqrt(2)*omega*sin(theta)/2*N.y + sqrt(2)*omega*cos(theta)/2*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y-A.z)
+    assert expand_mul(N.x.angle_between(A.x + A.y - A.z)) == 0 #Axis are aligned
+    assert (A.x + A.y - A.z).express(N).simplify() == sqrt(3)*N.x
+    assert A.dcm(N).simplify() == Matrix([[sqrt(3)/3, -sqrt(6)*sin(theta + pi/4)/3, sqrt(6)*cos(theta + pi/4)/3], \
+        [sqrt(3)/3, sqrt(6)*cos(theta + pi/12)/3, sqrt(6)*sin(theta + pi/12)/3], \
+        [-sqrt(3)/3, sqrt(6)*cos(theta + 5*pi/12)/3, sqrt(6)*sin(theta + 5*pi/12)/3]])
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y - omega*A.z)/sqrt(3)
+    assert A.ang_vel_in(N).magnitude() == Abs(omega)
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x + A.y-A.z)) == acos(omega/Abs(omega))
+    assert C.masscenter.vel(N).simplify() == (omega * A.y + omega * A.z)/sqrt(3)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == \
+        (1 - sqrt(3)/3)*N.x + sqrt(6)*sin(theta + pi/4)/3*N.y - sqrt(6)*cos(theta + pi/4)/3*N.z
+    assert C.masscenter.vel(N).express(N).simplify() == sqrt(6)*omega*cos(theta + pi/4)/3*N.y + sqrt(6)*omega*sin(theta + pi/4)/3*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    m, n = symbols('m n')
+    PinJoint('J', P, C, parent_joint_pos=m*N.x, child_joint_pos=n*A.x, child_axis=A.x+A.y-A.z, parent_axis=N.x-N.y+N.z)
+    assert expand_mul((N.x-N.y+N.z).angle_between(A.x+A.y-A.z)) == 0 #Axis are aligned
+    assert (A.x-A.y+A.z).express(N).simplify() == \
+        (-4*cos(theta)/3 - 1/3)*N.x + (1/3 - 4*sin(theta + pi/6)/3)*N.y + (4*cos(theta + pi/3)/3 - 1/3)*N.z
+    assert A.dcm(N).simplify() == Matrix([[S(1)/3 - 2*cos(theta)/3, -2*sin(theta + pi/6)/3 - S(1)/3, \
+        2*cos(theta + pi/3)/3 + S(1)/3], [2*cos(theta + pi/3)/3 + S(1)/3, 2*cos(theta)/3 - S(1)/3, \
+        2*sin(theta + pi/6)/3 + S(1)/3], [-2*sin(theta + pi/6)/3 - S(1)/3, 2*cos(theta + pi/3)/3 + S(1)/3, \
+        2*cos(theta)/3 - S(1)/3]])
+    assert A.ang_vel_in(N) == (omega*N.x - omega*N.y + omega*N.z)/sqrt(3)
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y - omega*A.z)/sqrt(3)
+    assert A.ang_vel_in(N).magnitude() == Abs(omega)
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x+A.y-A.z)) == acos(omega/Abs(omega))
```

Comment:
This should be zero. The angular velocity vector is aligned with the pin joint axis.

Reply:
Child axis is: `child_axis=A.x+A.y-A.z`

Reply:
yes it is zero, this the case I talked about in meeting, `omega/Abs(omega)` doesn't evaluate to 1.

Reply:
If you are setting the angular velocity to `omega*pin_axis_vec` then why would this have an `Abs()`?

Reply:
> If you are setting the angular velocity to `omega*pin_axis_vec` then why would this have an `Abs()`?

Because magnitude takes square root of it's square, value being real is returned with `Abs`

Reply:
Yes, I was incorrectly thinking that `.magnitude()` should be providing the magnitude and direction (in terms of the sign). This begs the question as to why we don't have a method to give the signed vector magnitude. This is fine to leave. Thanks.